Shared/API/CloudFunction.swift

```diff
@@ -53,22 +56,38 @@ extension CloudFunction {
 
             try Task.checkCancellation()
 
-            weakStatusables.forEach { (statusable) in
-                statusable.value?.handleEvent(status: .saved)
+            await withTaskGroup(of: Void.self) { group in
+                for weakStatusable in weakStatusables {
+                    group.addTask {
+                        guard let statusable = weakStatusable.value else { return }
+                        await statusable.handleEvent(status: .saved)
+                    }
+                }
```

Comment:
This taskGroup code boilerplate is a bit verbose and maybe a little confusing to read. I may make a convenience function in the future but I'll need to do a little more research first.